Modules/Order/Http/Controllers/OrderController.php

```diff
@@ -290,4 +363,197 @@ public function csv(Request $request): JsonResponse
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->orderService->processReturn($request->all());
+
+        return response()->json($response);
+    }
+
+    /**
+     * Initiate a order return in storage
+     *
+     * @param StoreReturnRequest $request
+     * @return JsonResponse
+     */
+    public function returnOrder(StoreReturnRequest $request): JsonResponse
+    {
+        $user = auth()->user();
+        $order = Order::where('order_number', $request->order_number)->first();
+
+        if ($user->cannot('update', $order)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->orderService->createReturnOrder($request->validated());
+
+        return response()->json($response);
+    }
+
+    /**
+     * Cancel order request by retailer
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function cancelRequest(Request $request): JsonResponse
+    {
+        $user = auth()->user();
+        $order = Order::where('order_number', $request->order_number)->first();
+
+        if ($user->cannot('cancelRequest', $order)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->orderService->cancelRequest($request);
+
+        return response()->json($response);
+    }
+
+    /**
+     * Add payment method by retailer
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function addPayment(Request $request): JsonResponse
+    {
+        $user = auth()->user();
+        $order = Order::where('order_number', $request->order_number)->first();
+
+        if ($user->cannot('cancelRequest', $order)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->orderService->addPayment($request);
+
+        return response()->json($response);
+    }
+
+    /**
+     * decline order from sent invoice by retailer
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function declineOrder(Request $request): JsonResponse
+    {
+        $user = auth()->user();
+        $order = Order::where('order_number', $request->order_number)->first();
+
+        if ($user->cannot('cancelRequest', $order)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->orderService->declineOrder($request);
```

Comment:
@arijit1111 cannot find the method `declineOrder`

Reply:
Please check in the OrderService file here: https://github.com/properweb/Bazar_Repo/blob/2fad6196663258d46b88d7762eecaa424d5038fd/Modules/Order/Http/Services/OrderService.php

Reply:
removed form route